code/controllers/subsystem/materials.dm

```diff
@@ -38,3 +41,22 @@ SUBSYSTEM_DEF(materials)
 	if(!materials)
 		InitializeMaterials()
 	return materials[fakemat] || fakemat
+
+///Returns a list to be used as an object's custom_materials. Lists will be cached and re-used based on the parameters.
+/datum/controller/subsystem/materials/proc/FindOrCreateMaterialCombo(list/materials_declaration, multiplier)
+	if(!material_combos)
+		InitializeMaterials()
+	var/list/combo_params = list()
+	for(var/x in materials_declaration)
+		var/datum/material/mat = x
+		var/path_name = ispath(mat) ? "[mat]" : "[mat.type]"
+		combo_params += "[path_name]=[materials_declaration[mat] * multiplier]"
+	sortTim(combo_params, /proc/cmp_text_asc) // We have to sort now in case the declaration was not in order
+	var/combo_index = combo_params.Join("-")
+	var/list/combo = material_combos[combo_index]
+	if(!combo)
+		combo = list()
```

Comment:
this wont save this list to material_combos

Reply:
it's set on line 61